hatch_build.py

```diff
@@ -53,3 +60,53 @@ def finalize(
         Clean up any generated temporary files.
         """
         self.known_implementations().unlink()
+
+    def _known_implementations_from_github(
+        self,
+        known_local: list[str],
+    ) -> list[str]:
+        """
+        Collects available implementation using GitHub repositories API.
+
+        The method uses GitHub repositories API to filter our repositories
+        marked with 'bowtie-harness' topic.
+        """
+        gh_token = os.getenv("GITHUB_TOKEN")
+
+        if not gh_token:
+            self.app.display_warning(
+                "GITHUB_TOKEN is not provided. "
+                "You can reach the GitHub rate limit",
+            )
+
+        # once all harnesses are moved we should report an error
+        # if collected list is empty
+        harnesses = self._collect_harnesses(gh_token)
+
+        self.app.display_debug(
+            f"Collected {len(harnesses)} harness(es): {harnesses}",
+        )
+
+        harnesses_set = set(harnesses)
+        local_set = set(known_local)
+        if harnesses_set.issubset(local_set):
+            return known_local
+
+        # join with local implementation until fully migrated to packages API
+        # https://github.com/bowtie-json-schema/bowtie/issues/1849
+        all_harnesses = list(local_set.union(harnesses_set))
+        # sorting is done just to simplify debugging
+        # if some harnesses are missed for some reason
+        all_harnesses.sort()
+        return all_harnesses
+
+    @staticmethod
+    def _collect_harnesses(gh_token) -> list[str]:
```

Comment:
note: I have tested this locally - the `kotlin-kmp-json-schema-validator` is returned as a collected test harness